### PR TITLE
Document when to use pluginFfi vs plugin

### PR DESCRIPTION
**Pull request by dcharkes**
Adds documentation from the questions answered in:

* https://github.com/flutter/flutter/issues/108974

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/packages-and-plugins/developing-packages.md [hunk 1]
```diff
@@ -614,6 +614,15 @@ a section in [Supporting the new Android plugins APIs][].
 If you want to develop a package that calls into native APIs using
 Dart's FFI, you need to develop an FFI plugin package.
 
+Both FFI plugin packages and (non-FFI) plugin packages support
+bundling native code, but FFI plugin packages do not support
+method channels and do include method channel registration code.
+If you want to implement a plugin that uses both method channels
+and FFI, use a (non-FFI) plugin. You can chose this per platform.
```
Comment (domesticmouse) [inline src/development/packages-and-plugins/developing-packages.md:621]: The last sentence probably needs a bit of rework to make it accessible to all our developers. /cc @sfshaza2

File: src/development/packages-and-plugins/developing-packages.md [hunk 2]
```diff
@@ -614,6 +614,16 @@ a section in [Supporting the new Android plugins APIs][].
 If you want to develop a package that calls into native APIs using
 Dart's FFI, you need to develop an FFI plugin package.
 
+Both FFI plugin packages and (non-FFI) plugin packages support
+bundling native code, but FFI plugin packages do not support
+method channels and do include method channel registration code.
+If you want to implement a plugin that uses both method channels
+and FFI, use a (non-FFI) plugin. You can chose per platform to
```
Comment (sfshaza2) [inline src/development/packages-and-plugins/developing-packages.md:621]: Hi, @dcharkes! Do you mean that if you might use FFI for, say, mobile, but non-FFI for, say, web, that you should implement this as federated plugin? I do want to tweak the wording, but I am not sure what you mean.

Comment (dcharkes) [reply src/development/packages-and-plugins/developing-packages.md:621]: One can implement mix and match (non-FFI e.g. method channel) plugins with FFI plugins for the native platforms.

See https://github.com/dcharkes/flutter/blob/91ef6ee700b9e281539e7734f5974bc5184894fa/packages/flutter_tools/lib/src/plugins.dart#L45-L63

**Comment by dcharkes** [comment]
@sfshaza2 did you mean this one?

```dart
  ///     flutter:
  ///      plugin:
  ///        platforms:
  ///          android:
  ///            package: io.flutter.plugins.sample
  ///            pluginClass: SamplePlugin
  ///          ios:
  ///            # A plugin implemented through method channels.
  ///            pluginClass: SamplePlugin
  ///          linux:
  ///            # A plugin implemented purely in Dart code.
  ///            dartPluginClass: SamplePlugin
  ///          macos:
  ///            # A plugin implemented with `dart:ffi`.
  ///            ffiPlugin: true
  ///          windows:
  ///            # A plugin using platform-specific Dart and method channels.
  ///            dartPluginClass: SamplePlugin
  ///            pluginClass: SamplePlugin
```
https://github.com/dcharkes/flutter/blob/91ef6ee700b9e281539e7734f5974bc5184894fa/packages/flutter_tools/lib/src/plugins.dart#L45-L63

**Review by sfshaza2 — APPROVED** [review]
Landing this for now, though I did want to tweak a bit.  But I'd like to land before freezing the website.

**Pull request merged**